File: src/01-easy/unary/solution.cpp
#include <iostream>
#include <string>
using namespace std;
int main() {
    string msg;
    getline(cin, msg);
    string bin;
    for (char c : msg)
        for (int b = 6; b >= 0; b--)
            bin += ((c >> b) & 1) ? '1' : '0';
    bool first = true;
    int i = 0;
    while (i < (int)bin.size()) {
        char cur = bin[i];
        int count = 0;
        while (i < (int)bin.size() && bin[i] == cur) { count++; i++; }
        if (!first) cout << " ";
        first = false;
        cout << (cur == '1' ? "0" : "00") << " " << string(count, '0');
    }
    cout << endl;
}

File: src/01-easy/unary/solution.fs
open System
let msg = Console.ReadLine()
let bin = msg |> Seq.map (fun c -> Convert.ToString(int c, 2).PadLeft(7, '0')) |> String.concat ""
let mutable i = 0
let mutable first = true
while i < bin.Length do
    let cur = bin.[i]
    let mutable count = 0
    while i < bin.Length && bin.[i] = cur do
        count <- count + 1
        i <- i + 1
    if not first then printf " "
    first <- false
    printf "%s %s" (if cur = '1' then "0" else "00") (String('0', count))
printfn ""
